lib/Basics/VelocyPackHelper.cpp

```diff
@@ -387,6 +391,188 @@ int VelocyPackHelper::compareNumberValues(VPackValueType lhsType,
   return (l < r ? -1 : 1);
 }
 
+template<typename T>
+int comp(T a, T b) {
+  if (a == b) {
+    return 0;
+  }
+  return (a < b ? -1 : 1);
+}
+
+static int compareInt64UInt64(int64_t i, uint64_t u) {
+  if (i < 0) {
+    return -1;
+  }
+  return comp<uint64_t>(static_cast<uint64_t>(i), u);
+}
+
+constexpr uint64_t uint64_2_52 = uint64_t{1} << 52;
+constexpr uint64_t uint64_2_63 = uint64_t{1} << 63;
+constexpr double double_2_52 = static_cast<double>(uint64_2_52);
+const double double_2_64 = ldexp(1.0, 64);
+
+static int compareUInt64Double(uint64_t u, double d) {
+  if (std::isnan(d)) {
```

Comment:
I think it's reasonable to specify here:
```suggestion
  if (std::isnan(d)) [[unlikely]] {
```
and similar for other NaN checks.

Reply:
done.